File: scripts/diagnostico-git.mjs
import { execSync } from 'child_process';
import { existsSync } from 'fs';
import path from 'path';

const CWD = '/vercel/share/v0-project';

function run(cmd) {
  console.log(`\n[v0] Executando: ${cmd}`);
  try {
    const out = execSync(cmd, { cwd: CWD, encoding: 'utf8', stdio: 'pipe' });
    console.log(`[v0] OK: ${out.trim() || '(sem output)'}`);
    return { ok: true, out };
  } catch (e) {
    console.log(`[v0] ERRO stdout: ${e.stdout?.trim() || '(vazio)'}`);
    console.log(`[v0] ERRO stderr: ${e.stderr?.trim() || '(vazio)'}`);
    console.log(`[v0] Exit code: ${e.status}`);
    return { ok: false, err: e };
  }
}

console.log('[v0] === DIAGNOSTICO GIT INICIADO ===');
console.log(`[v0] CWD existe: ${existsSync(CWD)}`);

// 1. Verificações básicas
run('git --version');
run('git remote -v');
run('git branch --show-current');
run('git status --short');
run('git log --oneline -3');

// 2. Stage e commit
run('git add -A');
run('git commit -m "FASE 4 COMPLETO: Redesign React profissional" || echo "Nada para commitar"');

// 3. Push
console.log('\n[v0] === TENTANDO PUSH ===');
const push = run('git push origin progresso-do-projeto 2>&1');

if (push.ok) {
  console.log('[v0] PUSH SUCESSO!');
} else {
  console.log('[v0] Push falhou. Verificando config de auth...');
  run('git config --list');
  run('cat /vercel/share/v0-project/.git/config');
}

console.log('\n[v0] === FIM DIAGNOSTICO ===');

File: scripts/diagnostico-git.py
import subprocess
import os
import sys

def run(cmd):
    print(f"\n[v0] Executando: {cmd}")
    result = subprocess.run(cmd, shell=True, capture_output=True, text=True, cwd="/vercel/share/v0-project")
    print(f"[v0] STDOUT: {result.stdout.strip() or '(vazio)'}")
    print(f"[v0] STDERR: {result.stderr.strip() or '(vazio)'}")
    print(f"[v0] Exit code: {result.returncode}")
    return result

print("[v0] === DIAGNOSTICO GIT INICIADO ===")
print(f"[v0] Python: {sys.version}")
print(f"[v0] CWD: {os.getcwd()}")

# Verifica se git existe
run("which git")
run("git --version")

# Verifica status do repo
run("git status")

# Verifica remote
run("git remote -v")

# Verifica branch atual
run("git branch --show-current")

# Verifica se tem arquivos novos
run("git status --short")

# Tenta adicionar arquivos
r = run("git add -A")
if r.returncode != 0:
    print("[v0] ERRO ao dar git add")
    sys.exit(1)

# Tenta commit
r = run('git commit -m "FASE 4 COMPLETO: Redesign React profissional com Next.js, Tailwind, Supabase"')
print(f"[v0] Commit resultado: {r.returncode}")

# Verifica se tem commits para fazer push
run("git log --oneline -5")

# Tenta push
r = run("git push origin progresso-do-projeto")
if r.returncode == 0:
    print("\n[v0] === PUSH FEITO COM SUCESSO! ===")
else:
    print(f"\n[v0] === FALHOU NO PUSH, tentando HEAD ===")
    r2 = run("git push")
    if r2.returncode == 0:
        print("[v0] Push com HEAD funcionou!")
    else:
        print("[v0] Ambos falharam. Verificando credenciais...")
        run("git config --list | grep -i url")
        run("git config --list | grep -i credential")

print("\n[v0] === DIAGNOSTICO CONCLUIDO ===")
